gcp/table_gcp_compute_tpu.go

```diff
@@ -0,0 +1,252 @@
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "network",
+				Description: "The name of the network that the TPU node is connected to.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "cidr_block",
+				Description: "The CIDR block that the TPU node will use when selecting an IP address.",
+				Type:        proto.ColumnType_CIDR,
+			},
+			{
+				Name:        "service_account",
+				Description: "The service account used to run the TPU node.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "create_time",
+				Description: "The time when the TPU node was created.",
+				Type:        proto.ColumnType_TIMESTAMP,
+			},
+			{
+				Name:        "scheduling_config",
+				Description: "Sets the scheduling options for the TPU instance.",
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "network_endpoints",
+				Description: "The network endpoints where the TPU node can be accessed.",
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "labels",
+				Description: "Resource labels to represent user provided metadata.",
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "zone",
+				Description: "The GCP zone where the TPU node is located.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromP(gcpTpuTurbotData, "Zone"),
+			},
+
+			// Tags
+			{
+				Name:        "tags_src",
+				Description: "A list of tags attached to the TPU node.",
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("Labels"),
+			},
+			{
+				Name:        "tags",
+				Description: "A map of tags for the resource.",
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.From(gcpTpuTags),
+			},
+
+			// Standard steampipe columns
+			{
+				Name:        "title",
+				Description: ColumnDescriptionTitle,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Name").Transform(lastPathElement),
+			},
+			{
+				Name:        "akas",
+				Description: ColumnDescriptionAkas,
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromP(gcpTpuTurbotData, "Akas"),
+			},
+
+			// Standard GCP columns
+			{
+				Name:        "project",
+				Description: "The GCP project ID.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromP(gcpTpuTurbotData, "Project"),
+			},
+		},
+	}
+}
+
+//// LIST FUNCTION
+
+func listComputeTpus(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("listComputeTpus")
+
+	// Create Service Connection
+	service, err := TPUService(ctx, d)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get project details
+	projectId, err := getProject(ctx, d, h)
+	if err != nil {
+		return nil, err
+	}
+
+	zone := d.EqualsQualString(matrixKeyZone)
+	parent := "projects/" + projectId.(string) + "/locations/" + zone
```

Comment:
Have you tried using the input parameter `projects/parker-aaa/locations/-`?

* This approach avoids the need to loop through each zone individually, which can be time-consuming due to separate API calls per zone.
* Using `-` after `locations/` retrieves all TPUs across all regions in a single request.